keps/sig-scheduling/1923-try-nominated-node-first/README.md

```diff
@@ -0,0 +1,224 @@
+pod to be scheduled, the selected node for the incoming pod after the
+preemption will be set as the value of `pod.Status.NominatedNodeName`.
+
+The Node is called *Nominated* to indicate the intent for the Pod to be
+scheduled on it once preemption of other Pods finish. However, the
+`Pod.status.nominatedNodeName` information is not directly used in the
+following scheduling cycles.
+
+Pod scheduling is split into two phases, the scheduling cycle and the binding
+cycle, the scheduling cycle primarily includes filtering and scoring.
+
+When the preemption happened in a previous scheduling cycle, there is a high
+chance that the nominated node is the *only* node that satisfies the filters
+for the incoming Pod to be scheduled.
+
+This KEP proposes to validate the availability of the nominated node in the
+new scheduling cycle before looping all over the nodes in the cluster for the
+phase of the filtering.
+
+## Motivation
+
+In real production environment, pods can have different priorites due to business
+needs, the preemption could happen to make sure high priority pods could got scheduled
+and spawned.
+
+For the cluster with huge number of computing nodes, looping over all the nodes in
+the cluster for the phase of the filtering is an unnegligible consuming for scheduler.
+
+### Goals
+
+- In the case of `pod.Status.NominatedNodeName` is set for the incoming pod, scheduler
+  will try the nominated node first for the filtering phase, failed to schedule on the
+  nominated node will fall back to verify all the nodes in the cluster instead of preemption
+  again.
+
+## Proposal
+
+### User Stories (Optional)
+
+As a premium user, I want my high priority application got spin up more quickly,
+I don't care where the application is running, whether there is any other nodes
+in the cluster have more disk space, memory size etc., as long as my application
+run normally.
+
+### Notes/Constraints/Caveats (Optional)
+
+When this feature is enabled the preemptor Pod might not be dispatched to the best
+candidated node in some corner case, e.g. another node releases the resources and
+becomes the best candidate while the victim pods got removed from the nominated node.
+
+## Design Details
+
+This feature hopes to improve the performance with the minimal change on the existing scheduling
+framework, scheduler will short-circuit to check the nominated node directly instead of looping
+all the nodes in the cluster, and if the nominated node doesn't fit, a fresh new scheduling cycle
+will be started to get an new node for scheduling.
+
+In most situation, the new scheduling cycle is not needed, but we still got the case like the
+topology changing, victim pods deletion is still on-the-fly in the following scheduling cycle.
+Those cases will make the nominated node not fits for the coming pod anymore, and a new scheduling
+cycle is needed in those case. This is no harm since it will eventually get convergent.
```

Comment:
I would simplify this as: 

"We will update the scheduling cycle such that nominated node of a pod is evaluated first and schedule the pod on that node if it fits. If the nominated node doesn't fit the pod, only then the scheduling cycle continues with the standard logic of evaluating the rest of the nodes in the cluster."

Reply:
we already mentioned that two times in the summary and goals, I feel we repeated the same thing multiple times unnecessarily. So I think we can even remove this section completely and directly get into the implementation details.

Reply:
removed.